### PR TITLE
Register config file with Webpack for watching

### LINKED ISSUE #55: Triggering rebuild on config.js change
Enjoying working with the release, good work!

I'd like to propose triggering a rebuild when the config file (`tailwind-config.js`) is updated if a build tool is being used – ie. webpack. 

Basically in the same way as a new build is triggered when you update a css/sass/js file. It would mean that modifications can be made to the config file without having to halt and restart `npm watch` for instance.

### PR DESCRIPTION
**Pull request by adamwathan**
This change tells Webpack (for those using it) that the config file used should be tracked as a dependency and that the CSS should be re-compiled if that config file changes.

It's careful to make sure the config file is loaded fresh every single time to avoid weird caching issues.

Resolves #55 🙌

### CODE REVIEW

File: docs/tailwind.js [hunk 1]
```diff
@@ -1,4 +1,5 @@
-var config = require('../defaultConfig')
+var config = require('../lib/index').defaultConfig()
+
```
Comment (reinink) [inline docs/tailwind.js]: Was this extra line return really necessary Adam?

**Review by reinink — APPROVED** [review]
proceed

File: docs/tailwind.js (hunk 1 above)
Comment (adamwathan) [reply docs/tailwind.js]: Sorry dad, fixed.

Comment (reinink) [reply docs/tailwind.js]: thank you

**Review by vcavallo — COMMENTED** [review]
I found this old thread and it made me very happy.

**Comment by anantakrishna** [comment]
Hi guys. Any idea how to make it work with https://github.com/parcel-bundler/parcel v1.12? Currently I have to remove [cache directory](https://en.parceljs.org/cli.html#cache-directory) in order for it to catch the changes and re-bundle CSS.

Oh, before posting this comment I've found a related Parcel issue https://github.com/parcel-bundler/parcel/issues/2983, and it turns out that the pull request https://github.com/parcel-bundler/parcel/pull/3980 was merged into Parcel 2.0 (not 1.x).

**Comment by leoncvlt** [comment]
> Hi guys. Any idea how to make it work with https://github.com/parcel-bundler/parcel v1.12? Currently I have to remove [cache directory](https://en.parceljs.org/cli.html#cache-directory) in order for it to catch the changes and re-bundle CSS.
> 
> Oh, before posting this comment I've found a related Parcel issue [parcel-bundler/parcel#2983](https://github.com/parcel-bundler/parcel/issues/2983), and it turns out that the pull request [parcel-bundler/parcel#3980](https://github.com/parcel-bundler/parcel/pull/3980) was merged into Parcel 2.0 (not 1.x).

I was trying to fix this same issue, in the end I created a separate build command in my package.json that start the development server with no cache `parcel serve src/html/index.html --no-cache --open` - still decently fast, and the changes to `tailwind.config.js` are picked up, I do have to manually refresh the browser though.

**Comment by leoncvlt** [comment]
> Hi guys. Any idea how to make it work with https://github.com/parcel-bundler/parcel v1.12? Currently I have to remove [cache directory](https://en.parceljs.org/cli.html#cache-directory) in order for it to catch the changes and re-bundle CSS.
> 
> Oh, before posting this comment I've found a related Parcel issue [parcel-bundler/parcel#2983](https://github.com/parcel-bundler/parcel/issues/2983), and it turns out that the pull request [parcel-bundler/parcel#3980](https://github.com/parcel-bundler/parcel/pull/3980) was merged into Parcel 2.0 (not 1.x).

Trying to achieve the same thing, in the end I made a separate development start command that runs parcel without the cache: `parcel serve src/html/index.html --no-cache --open` - it's still decently fast - you'll have to change a `.css` file after making your changes to `tailwind.config.js` to trigger css rebuilding.

**Pull request merged**